src/k8s/pkg/k8sd/setup/k8s_apiserver_proxy.go

```diff
@@ -20,7 +20,7 @@ func K8sAPIServerProxy(snap snap.Snap, servers []string, localhostAddress string
 	if _, err := snaputil.UpdateServiceArguments(snap, "k8s-apiserver-proxy", map[string]string{
 		"--endpoints":  configFile,
 		"--kubeconfig": filepath.Join(snap.KubernetesConfigDir(), "kubelet.conf"),
-		"--listen":     fmt.Sprintf("%s:6443", localhostAddress),
+		"--listen":     fmt.Sprintf(":%s", securePort),
```

Comment:
This uses the default interface instead of the specified localhost, right?
I guess this will fail if we run a ipv6-only cluster on a cluster with both interfaces as IPv4 is preferred.

Reply:
AFAIK It will listen to all interfaces.

https://pkg.go.dev/net#Listen
https://pkg.go.dev/net#Dial
Has some more information related to this.

Reply:
Fair point, I thought this works similar to the `kubelet` `--node-ip` flag which will prefer the Ipv4 interface if you not set `::` 

But you are right, that is something that we implemented, not upstream.